sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -50,25 +50,30 @@ def lambda_handler(event, context):
 
     :return: A pre-signed url where users can upload a data file to the SDS.
     """
-    logger.info(f"Event: {event}")
-    logger.info(f"Context: {context}")
+    path_params = event.get("pathParameters", {}).get("proxy", None)
+    logger.debug("Parsing path parameters=[%s] from event=" "[%s]", path_params, event)
 
-    if "filename" not in event["queryStringParameters"]:
+    if not path_params:
         return {
             "statusCode": 400,
-            "body": json.dumps("Please specify a filename to upload"),
+            "body": json.dumps(
+                "No filename given for upload. "
+                "Please provide a filename "
+                "in the path. Eg. /upload/filename.pkts"
+            ),
         }
-
-    filename = event["queryStringParameters"]["filename"]
-
-    science_file = ScienceFilepathManager(filename)
-
-    if not science_file.is_valid:
-        return {"statusCode": 400, "body": science_file.error_message}
+    # TODO: Handle other filetypes other than science files
+    #      The current ScienceFilepathManager only accepts filenames, not the full path
+    filename = os.path.basename(path_params)
+    try:
+        science_file = ScienceFilepathManager(filename)
+    except InvalidScienceFileError as e:
+        logger.error(str(e))
+        return {"statusCode": 400, "body": str(e)}
 
     s3_key_path = science_file.construct_upload_path()
 
-    url = _generate_signed_upload_url(s3_key_path, tags=event["queryStringParameters"])
+    url = _generate_signed_upload_url(s3_key_path)
```

Comment:
Recently, I noticed that we(including myself) don't add as many logs for when everything goes as expected and that leads to almost empty logs on cloudwatch which makes it seems like nothing happened.

Reply:
You want to include tags on the s3 object? The queryStringParameters should be what was sent in the pathParams now, which was logged on entry to the lambda I thought.

Reply:
Sorry for the confusion. This comment was not meant for this line of code but as general comment.